src/amltk/optimization/optimizer.py

```diff
@@ -123,3 +130,92 @@ def preferred_parser(
 
         """
         return None
+
+    @classmethod
+    @abstractmethod
+    def create(
+        cls,
+        *,
+        space: Node,
+        metrics: Metric | Sequence[Metric],
+        bucket: str | Path | PathBucket | None = None,
+        seed: Seed | None = None,
+    ) -> Self:
+        """Create this optimizer.
+
+        !!! note
+
+            Subclasses should override this with more specific configuration
+            but these 3 arguments should be all that's necessary to create
+            the optimizer.
+
+        Args:
+            space: The space to optimize over.
+            bucket: The bucket for where to store things related to the trial.
+            metrics: The metrics to optimize.
+            seed: The seed to use for the optimizer.
+
+        Returns:
+            The optimizer.
+        """
+
+    class CreateSignature(Protocol):
```

Comment:
Note it's a `Protocol`, akin to a Java interface. I use it for type hinting what a function signature should look like, i.e. any function that looks like `Optimizer.create`